apiservice/proxy.go

```diff
@@ -0,0 +1,249 @@
+package apiservice
+
+import (
+	"context"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/onflow/flow/protobuf/go/flow/access"
+
+	"github.com/onflow/flow-go/engine/access/rpc/backend"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/utils/grpcutils"
+)
+
+func NewFlowProxy(accessNodeAddressAndPort flow.IdentityList, timeout time.Duration) (*FlowProxy, error) {
+	accessClients := make([]access.AccessAPIClient, accessNodeAddressAndPort.Count())
+	for i, identity := range accessNodeAddressAndPort {
+		clientRPCConnection, err := grpc.Dial(
+			identity.Address,
+			grpc.WithDefaultCallOptions(grpc.MaxCallRecvMsgSize(grpcutils.DefaultMaxMsgSize)),
+			grpc.WithInsecure(), //nolint:staticcheck
+			backend.WithClientUnaryInterceptor(timeout))
+		if err != nil {
+			return nil, err
+		}
+
+		accessClients[i] = access.NewAccessAPIClient(clientRPCConnection)
+	}
+
+	ret := &FlowProxy{}
+	ret.upstream = accessClients
+	ret.LocalCache = nil
+	ret.roundRobin = 0
+	return ret, nil
+}
+
+type FlowProxy struct {
+	access.UnimplementedAccessAPIServer
```

Comment:
It makes sense. Some of this is experimental, since we wanted to extend this with the Protocol API. Let me look into this. Running a full test run usually helps to assess if it is right.

Reply:
I am working on this one now.

Reply:
You first option proved to be the shortest and most straightforward. I am fixing some lint issues but I will push that change once all tests pass.

Reply:
Fixed: https://github.com/GetElastech/flow-go/commit/61fae7fc15bed9aa3969d55ea1af704428ddc9ef